django/db/backends/base/base.py

```diff
@@ -738,6 +738,20 @@ def on_commit(self, func):
             # immediately.
             func()
 
+    def on_commit_robust(self, func):
+        if not callable(func):
+            raise TypeError("on_commit()'s callback must be a callable.")
+        if self.in_atomic_block:
+            # Transaction in progress; save for execution on commit.
+            self.run_on_commit.append((set(self.savepoint_ids), func))
```

Comment:
When we register the function to run robustly on commit in an atomic bloc, should we also try/catch it when it is actually run?
Meaning that in `db.backends.base.base.BaseDatabaseWrapper.run_and_clear_commit_hooks` we should take into account this info

Reply:
Can you please elaborate more of what you expect here?

Reply:
When calling `on_commit` there are basically two modes:
- there is no transaction in progress, so we execute the function right away
- we are in an atomic block, so we register the function to execute it later (`self.run_on_commit.append(`)

The first case is handled by the PR, but not the second one. And I'd think that we would need to handle a robust execution in the second case too.
Does that make it clearer? :)

Reply:
Yes! Thanks for the reply. I will commit the changes soon.